File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/metaschema/Contribute.ts
import {IntervalloTemporale} from '../anagrafiche/IntervalloTemporale';

export class Contribute {
  code: number;
  type: number;
  description: string;
  validity: IntervalloTemporale;

  constructor(code: number = 0, type: number = 0) {
    this.code = code;
    this.type = type;
    this.description = '';
    this.validity = new IntervalloTemporale();
  }
}

export class LinkedContribute<T> extends Contribute {
  linkValidity: IntervalloTemporale;
  linkedItemPK: T;

  constructor(code: number = 0, type: number = 0, linkedItemPK: T = undefined) {
    super(code, type);
    this.linkedItemPK = linkedItemPK;
  }
}

export enum ContributeType {
  Anything = 0,
  ACA = 1
}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/metaschema/Contribute.cs
using AgronicaCoreModelsSTD.anagrafiche;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.metaschema
{
    public class Contribute
    {
        public int code { get; set; }
        public int type { get; set; }
        public string description { get; set; }
        public IntervalloTemporale validity { get; set; }

        public Contribute(int code, int type) 
        {
            this.code = code;
            this.type = type;
            this.description = "";
            this.validity = new IntervalloTemporale();
        }

        public Contribute() { }
    }

    public class LinkedContribute<T> : Contribute
    {
        public IntervalloTemporale linkValidity { get; set; }
        public T linkedItemPK { get; set; }

        public LinkedContribute(int code, int type, T linkedItemPK) : base(code, type)
        {
            this.linkedItemPK = linkedItemPK;
            this.linkValidity = new IntervalloTemporale();
        }

        public LinkedContribute() : base() { }
    }
}
